caas/kubernetes/provider/resources.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"fmt"
+
+	"github.com/juju/errors"
+	"github.com/juju/version"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/tags"
+)
+
+// AdoptResources is called when the model is moved from one
+// controller to another using model migration.
+func (k *kubernetesClient) AdoptResources(ctx context.ProviderCallContext, controllerUUID string, fromVersion version.Number) error {
+	modelLabel := fmt.Sprintf("%v==%v", tags.JujuModel, k.modelUUID)
+
+	pods := k.CoreV1().Pods(k.namespace)
+	podsList, err := pods.List(v1.ListOptions{
+		LabelSelector: modelLabel,
+	})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, p := range podsList.Items {
+		p.Labels[tags.JujuController] = controllerUUID
+		if _, err := pods.Update(&p); err != nil {
```

Comment:
The updates look good, but other provider implementations of AdoptResources *cough*Azure*cough* are a bit slow on big models - is there any way to do bulk updates for these?

Reply:
sadly the k8s api does not allow such things IIANM - you need to access each entity type via the specific endpoint for that entity type